jbpm-designer-client/src/main/java/org/jbpm/designer/client/shared/AssignmentData.java

```diff
@@ -351,6 +356,68 @@ protected void setDisallowedPropertyNames(String disallowedPropertyNames) {
         }
     }
 
+    public Map<String, List<String>> getCustomAssignmentProperties() {
+        return mapCustomAssignmentProperties;
+    }
+
+    protected void setCustomAssignmentProperties(final String customassignmentproperties) {
+        this.mapCustomAssignmentProperties.clear();
+        if (customassignmentproperties != null && !customassignmentproperties.isEmpty()) {
+            String[] caps = customassignmentproperties.split(",");
+            for (String cap : caps) {
+                if (!cap.isEmpty()) {
+                    String[] capParts = cap.split(":");
+                    if (capParts.length > 0 && !capParts[0].isEmpty()) {
+                        String propName = capParts[0].trim();
+                        List<String> listPropValues = new ArrayList<String>();
+                        if (capParts.length > 1) {
+                            String propValues = capParts[1].trim();
+                            String[] pvs = propValues.split(";");
+                            if (pvs.length > 0) {
+                                for (String pv : pvs) {
+                                    pv = pv.trim();
+                                    if (!pv.isEmpty()) {
+                                        listPropValues.add(pv);
+                                    }
+                                }
+                            }
+                        }
+                        this.mapCustomAssignmentProperties.put(propName,
+                                                               listPropValues);
+                    }
+                }
+            }
+        }
+    }
+
+    public String getCustomAssignmentPropertiesString() {
+        StringBuilder sb = new StringBuilder();
+        if (!mapCustomAssignmentProperties.isEmpty()) {
+            Iterator<String> itPropNames = mapCustomAssignmentProperties.keySet().iterator();
+            while (itPropNames.hasNext()) {
+                String propName = itPropNames.next();
+                sb.append(propName).append(':');
+                List<String> propValues = mapCustomAssignmentProperties.get(propName);
+                if (!propValues.isEmpty()) {
+                    Iterator<String> itPropValues = propValues.iterator();
+                    while (itPropValues.hasNext()) {
+                        String propValue = itPropValues.next();
+                        if (!propValue.isEmpty()) {
+                            sb.append(propValue).append(';');
+                        }
+                    }
+                }
+                sb.append(',');
+            }
+        }
+        return sb.toString();
+    }
```

Comment:
@hasys yeah same thing - i think could be reduced but parsing strings with couple separators is tricky